src/kolibri_tools/utils.py

```diff
@@ -0,0 +1,76 @@
+import json
+import logging
+import os
+from collections import Mapping
+
+from config import KOLIBRI_PORT
+
+
+def start_kolibri_server():
+    from kolibri.utils.cli import initialize, setup_logging, start
+    from kolibri.plugins.registry import registered_plugins
+
+    registered_plugins.register_plugins(['kolibri.plugins.app'])
+
+    logging.info("Starting server...")
+    setup_logging(debug=False)
+    initialize()
+    automatic_provisiondevice()
+    start.callback(KOLIBRI_PORT, background=False)
+
+
+def get_initialize_url(next_url=None):
+    from kolibri.utils.cli import initialize
+    from kolibri.plugins.registry import registered_plugins
+
+    # The start_kolibri_server function is typically run in a different thread or process
+    # than the app, so for get_initialize_url to work, we need to initialize Kolibri
+    # in the app process as well.
+    registered_plugins.register_plugins(['kolibri.plugins.app'])
+    initialize()
+
+    from kolibri.plugins.app.utils import interface
+
+    return interface.get_initialize_url(next_url)
+
+
+def get_content_file_path(filename):
+    from kolibri.core.content.utils.paths import get_content_storage_file_path
+    return get_content_storage_file_path(filename)
+
+
+def automatic_provisiondevice():
```

Comment:
Is this something we'd perhaps want in Kolibri proper, @rtibbles? Then it could be used for non-app installations as well.

Reply:
Yes, I think so - also means that this functionality should be less prone to breakage if something is changed in provision device.

Reply:
Going to file an issue for this in the Kolibri repo, and leave it as a next step.

Reply:
Issue filed! https://github.com/learningequality/kolibri/issues/7803